core/core.go

```diff
@@ -87,8 +93,10 @@ type NetworkDriver interface {
 	DeleteNetwork(id string) error
 	CreateEndpoint(id string) error
 	DeleteEndpoint(id string) error
-	CreatePeerHost(id string) error
-	DeletePeerHost(id string) error
+	AddPeerHost(node ServiceInfo) error
```

Comment:
minor:
I think you can just embed `Clustering` interface here as well unless you want to differentiate it from the one in `Plugin`